data-analytics/dataflow-python-examples/dataflow_python_examples/data_generation_for_benchmarking.py

```diff
@@ -249,35 +250,37 @@ def sanity_check(self, record, fieldname):
             if self.data_gen.only_pos and record[fieldname] < 0:
                 record[fieldname] = abs(record[fieldname])
             record[fieldname] = int(record[fieldname])
-        elif field[u'type'] == 'FLOAT':
+        elif field['type'] == 'FLOAT':
             # This implements max and sign constraints
-            record[fieldname] = faker.pyfloat(math.log10(self.data_gen.max_float),
-                                              self.data_gen.float_precision,
-                                              self.data_gen.only_pos)
-            record[fieldname] = float(record[fieldname])
+            if (record[fieldname] > self.data_gen.max_float or
+                    (record[fieldname] < 0 and self.data_gen.only_pos)):
+                record[fieldname] = faker.pyfloat(math.log10(self.data_gen.max_float),
+                                                  self.data_gen.float_precision,
+                                                  self.data_gen.only_pos)
+            record[fieldname] = round(float(record[fieldname]), self.data_gen.float_precision)
 
         # Make some values null based on null_prob.
-        if field[u'mode'] == 'NULLABLE':
+        if field['mode'] == 'NULLABLE':
             record[fieldname] = np.random.choice([None, record[fieldname]],
                                                  p=[self.data_gen.null_prob,
                                                     1.0 - self.data_gen.null_prob])
 
         # Pick key at random from foreign keys.
         # Draw key column from [0, n_keys) if has _key in the name.
         # This forces key column to no contain nulls
-        if ('_key' in field[u'name'].lower()) or ('_id' in field[u'name'].lower()):
+        if '_key' in field['name'].lower() or '_id' in field['name'].lower():
             key_mag = int(math.log10(self.data_gen.n_keys))
             # Assume the key field is of string type and format it to be left zero padded.
             record[fieldname] = format(np.random.randint(0, self.data_gen.n_keys),
-                                       '0' + str(key_mag + 1) + 'd')
+                                       '0%dd' % (key_mag + 1))
```

Comment:
Be careful with this, lots of languages treat numbers that start with `0` as octal, so `010` actually ends up as `8` in decimal.

Reply:
Noted. Changing to 'L-0%dd' to avoid octal confusion with key strings.